### ISSUE TITLE
Cardinal Mobile SDK update

### ISSUE THREAD

**Issue by noebt** [pikaju/flutter-braintree#129]
Hi! I have a flutter app using this package in the Play Store, and I just received this email from Google:

Your app is using the **Cardinal Mobile SDK** which collects persistent device identifiers, but may not be limited to Android ID and hardware IDs. Persistent device identifiers may not be linked to any other personal and sensitive user data and/or resettable device identifiers (e.g., IMEI, IMSI, SIM Serial #, etc).

You may consider upgrading to a policy-compliant version of this SDK, if available from your SDK provider or removing the SDK. According to your SDK provider, you may consider upgrading to **2.2.7-2**. Please consult the SDK provider for further information. Google is unable to endorse or recommend any third party software.

I looked for this and it seems that it's related to Braintree. Does anyone know what to do to solve this? Thanks!

**Comment by sharpsan** [comment]
I just got this emal, too.  I am using the latest stable version of this package.

**Comment by sujindra-maharjan** [comment]
I also got the same email.

**Comment by andersonfds** [comment]
Also got this email

**Comment by nicolobozzato** [comment]
As the email said we should use the new version of Cardinal at least 7.2.2-2. 
Version 4.18.0 of the Braintree SDK starts to implement that version, as we use Braintree Drop-In SDK we have to update it to version 6.4.0 that starts to use that version.

**Comment by noebt** [comment]
so what can we do in our side? would it be enough to update the dependency to 6.4.0 in the build.gradle file? Like     `implementation 'com.braintreepayments.api:drop-in:6.4.0'`

**Comment by noebt** [comment]
As quick fix something like this may work

```
implementation('com.braintreepayments.api:drop-in:5.2.1'){
      exclude group: "org.jfrog.cardinalcommerce.gradle", module: "cardinalmobilesdk"
}
implementation("org.jfrog.cardinalcommerce.gradle:cardinalmobilesdk:2.2.7-2")
```

**Comment by Anderillo** [comment]
We also got this email!

**Comment by iamar1k92** [comment]
any permanent fix?

**Comment by DavideM12** [comment]
> As quick fix something like this may work
> 
> ```
> implementation('com.braintreepayments.api:drop-in:5.2.1'){
>       exclude group: "org.jfrog.cardinalcommerce.gradle", module: "cardinalmobilesdk"
> }
> implementation("org.jfrog.cardinalcommerce.gradle:cardinalmobilesdk:2.2.7-2")
> ```

@noebt can you confirm us that it works? Does Google Play accept the update?

Thank you

**Comment by noebt** [comment]
> > As quick fix something like this may work
> > ```
> > implementation('com.braintreepayments.api:drop-in:5.2.1'){
> >       exclude group: "org.jfrog.cardinalcommerce.gradle", module: "cardinalmobilesdk"
> > }
> > implementation("org.jfrog.cardinalcommerce.gradle:cardinalmobilesdk:2.2.7-2")
> > ```
> 
> @noebt can you confirm us that it works? Does Google Play accept the update?
> 
> Thank you

I just submitted an updated version with this fix and it was accepted by Google Play, so I guess it works

**Comment by nicolobozzato** [comment]
Sorry if I ask again, but for you 3dSecure keeps working regularly?

**Comment by vkozinec** [comment]
We got the same email today, hopefully this will be addressed and fixes here https://github.com/pikaju/flutter-braintree/pull/131

**Comment by nicolobozzato** [comment]
@vkozinec 
With the solution of @noebt I had a problem with my implementation of 3DS, so I did my best to fix it with the PR linked, but at the price of deleting all the custom part that personally I didn't need, but somebody else may need (hence why it's only a draft).

**Comment by vkozinec** [comment]
@nicolobozzato 

Thanks for creating that PR, could you please just give us quick notes on what should not work in your implementation?

We use Braintree mainly on Android and iOS devices, not web, and we need GPay, Card and Apple Pay to work with 3DS secure, will that work then?

By deleting custom part, you are referencing to custom UI for Braintree?

**Comment by vkozinec** [comment]
Looks like this is fixed here https://github.com/braintree/braintree-android-drop-in/issues/373 and should be just a matter of updating it if I am not mistaken.

**Comment by nicolobozzato** [comment]
@vkozinec sorry for the late. Yes, it's indeed enough to increase the version of the SDK but as it is a breaking change, this breaks all the android 3ds implementations, so I rewrote that part.  IOS is not touched only android, I currently use Paypal and Credit Card so those are fairly tested. GPay should work, but I would make some test (sorry). Apple Pay if I'm not mistaken I don't think it was supported even before.

**Comment by vkozinec** [comment]
@nicolobozzato great, thanks for the response. I saw that there is conversation moving in PR, hopefully this will be merged soon so we do not have to fork and manually update things.

**Comment by pikaju** [comment]
This should hopefully be fixed in `4.0.0-dev.1`
